recipes/libusb/all/conanfile.py

```diff
@@ -53,100 +52,136 @@ def config_options(self):
-        del self.settings.compiler.libcxx
-        del self.settings.compiler.cppstd
+        try:
+            del self.settings.compiler.libcxx
+        except Exception:
+            pass
+        try:
+            del self.settings.compiler.cppstd
+        except Exception:
+            pass
+        self.win_bash = self._is_mingw
 
     def build_requirements(self):
-        if self._settings_build.os == "Windows" and not self._is_msvc and not tools.get_env("CONAN_BASH_PATH"):
-            self.build_requires("msys2/cci.latest")
+        if self._settings_build.os == "Windows" and not is_msvc(self) and not self.conf_info.get("tools.microsoft.bash:path"):
+            self.tool_requires("msys2/cci.latest")
 
     def requirements(self):
         if self.settings.os == "Linux":
             if self.options.enable_udev:
                 self.requires("libudev/system")
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version],
-                  destination=self._source_subfolder, strip_root=True)
-
-    def _build_visual_studio(self):
-        with tools.chdir(self._source_subfolder):
-            # Assume we're using the latest Visual Studio and default to libusb_2019.sln
-            # (or libusb_2017.sln for libusb < 1.0.24).
-            # If we're not using the latest Visual Studio, select an appropriate solution file.
-            solution_msvc_year = 2019 if tools.Version(self.version) >= "1.0.24" else 2017
-
-            solution_msvc_year = {
-                "11": 2012,
-                "12": 2013,
-                "14": 2015,
-                "15": 2017
-            }.get(str(self.settings.compiler.version), solution_msvc_year)
-
-            solution_file = os.path.join("msvc", "libusb_{}.sln".format(solution_msvc_year))
-            platforms = {"x86":"Win32"}
-            properties = {
-                # Enable LTO when CFLAGS contains -GL
-                "WholeProgramOptimization": "true" if any(re.finditer("(^| )[/-]GL($| )", tools.get_env("CFLAGS", ""))) else "false",
-            }
-            msbuild = MSBuild(self)
-            build_type = "Debug" if self.settings.build_type == "Debug" else "Release"
-            msbuild.build(solution_file, platforms=platforms, upgrade_project=False, properties=properties, build_type=build_type)
-
-    def _configure_autotools(self):
-        if not self._autotools:
-            self._autotools = AutoToolsBuildEnvironment(self, win_bash=tools.os_info.is_windows)
-            configure_args = ["--enable-shared" if self.options.shared else "--disable-shared"]
-            configure_args.append("--enable-static" if not self.options.shared else "--disable-static")
+        get(self, **self.conan_data["sources"][self.version],
+                  destination=self.source_folder, strip_root=True)
+
+    def layout(self):
+        if is_msvc(self):
+            vs_layout(self)
+        else:
+            basic_layout(self, src_folder="src")
```

Comment:
I'm not sure that conditional layout is a good idea, since it controls:
- source layout
- build layout
- package layout

But:
- source layout must be the same regardless of settings
- build layout is an internal detail so it's fine
- package layout must be the same regardless of settings

Reply:
Good question, I gonna open an issue to Conan devs about that case. Indeed is not something usual.